ios/Codeforces Watcher/Application/features/users/cells/userAccount/UserAccountCommonInfoView.swift

```diff
@@ -0,0 +1,148 @@
+//
+//  UserAccountCommonInfoView.swift
+//  Codeforces Watcher
+//
+//  Created by Ivan Karavaiev on 3/10/21.
+//  Copyright © 2021 xorum.io. All rights reserved.
+//
+
+import UIKit
+import common
+
+class UserAccountCommonInfoView: UIView {
+    
+    private let contentView = UIView()
+
+    private let avatarImage = CircleImageView().apply {
+        $0.width(48)
+        $0.height(48)
+    }
+    private let rankLabel = UILabel().apply {
+        $0.textColor = Palette.colorPrimary
+        $0.font = Font.textSubheadingBig
+    }
+    private let handleLabel = UILabel().apply {
+        $0.textColor = Palette.colorPrimary
+        $0.font = Font.textBody
+    }
+    private let nameLabel = UILabel().apply {
+        $0.font = Font.textSubheadingBig
+    }
+    private let lastUpdateLabel = UILabel().apply {
+        $0.textColor = Palette.gray
+        $0.font = Font.textSubheading
+    }
+    
+    struct UIModel {
+        let avatar: String
+        let rank: String?
+        let handle: String
+        let firstName: String?
+        let lastName: String?
+        let lastUpdate: Int64?
+    }
+    
+    private var uiModel: UserAccountCommonInfoView.UIModel?
+    
+    public override init(frame: CGRect) {
+        super.init(frame: frame)
+        setupView()
+    }
+
+    public required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        setupView()
+    }
+
+    private func setupView() {
+        clipsToBounds = false
+        buildViewTree()
+        setConstraints()
+    }
+
+    private func buildViewTree() {
+        addSubview(contentView)
+        [avatarImage, rankLabel, handleLabel, nameLabel, lastUpdateLabel].forEach(contentView.addSubview)
+    }
+
+    private func setConstraints() {
+        contentView.edgesToSuperview()
+        
+        avatarImage.run {
+            $0.topToSuperview(offset: 16)
+            $0.centerXToSuperview()
+        }
+        
+        rankLabel.run {
+            $0.topToBottom(of: avatarImage, offset: 8)
+            $0.centerXToSuperview()
+        }
+        
+        handleLabel.run {
+            $0.topToBottom(of: rankLabel, offset: 4)
+            $0.centerXToSuperview()
+        }
+        
+        nameLabel.run {
+            $0.topToBottom(of: handleLabel, offset: 8)
+            $0.centerXToSuperview()
+        }
+        
+        lastUpdateLabel.run {
+            $0.topToBottom(of: nameLabel, offset: 12)
+            $0.centerXToSuperview()
+            $0.bottomToSuperview(offset: -8)
+        }
+    }
+    
+    func bind(_ uiModel: UserAccountCommonInfoView.UIModel) {
+        self.uiModel = uiModel
+        
+        let avatar = LinkValidatorKt.avatar(avatarLink: uiModel.avatar)
```

Comment:
btw, we don't need to use LinkValidator here anymore, because it's already done on backend